<!DOCTYPE html>
<html lang="en">

<head><!-- Copyright by Martin Gühmann, Stefanie Reuter,  and Ralf Mrowka (C) 2024, all rights reserved -->
	<meta charset="utf-8">

	<meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate">
	<meta http-equiv="Pragma" content="no-cache">
	<meta http-equiv="Expires" content="0">

	<title>DiffDigester</title>
	<link rel="stylesheet" href="stylesheet.css" type="text/css" media="screen">
	<link rel="shortcut icon" href="favicon.ico" type="image/x-icon">
	<link rel="icon" href="favicon.ico" type="image/x-icon">

	<script var="" database;="" src="gcg.txt"></script>
	<script>var enzymeentry = database.split('\n');</script>
	<script type="text/javascript" src="digester.js"></script>
</head>

<body>
	<div class="w3-bar w3-black">
		<button class="w3-bar-item w3-button tablink w3-blue" onclick="pressed(event,'home')">Home</button>
		<button class="w3-bar-item w3-button tablink" onclick="pressed(event,'documentation')">Why Diff Digestion?</button>
		<button class="w3-bar-item w3-button tablink" onclick="pressed(event,'help')">Documentation and Help</button>
		<button class="w3-bar-item w3-button tablink" onclick="pressed(event,'digester')">Diff Digest</button>
		<button class="w3-bar-item w3-button tablink" onclick="pressed(event,'about')">About</button>
	</div>

	<div id="home" class="w3-container  tab">

		<figure style="float: right">
			<img src="gel-spiegel2.jpg" alt="gel image" style="width: 300px">
			<figcaption>Image of an DNA agarose gel</figcaption>
		</figure> 

		<noscript>
			<div class="h1" style="color:red">
				This site requires JavaScript, please enable it!
			</div>
		</noscript>

		<div class="h1">Welcome to DiffDigester</div>
		Molecular cloning is a basic and very useful method in molecular biology.
		Cloning allows combining DNA fragments from different sources into
		a single DNA molecule, such as a plasmid. Plasmids can be easily amplified
		in E. coli and also be used to express proteins in bacteria and
		eukaryotic cells. Plasmids can also be used, for example, as p-element
		vectors to create transgenic fruit flies.<Br>

		One application for molecular cloning is to replace a gene in a plasmid by
		another gene. Another is adding several functional elements into a plasmid.
		For example, a p-element vector has not only a plasmid backbone and
		the gene of interest, but also a promoter for tissue-specific expression,
		a 3'UTR with a polyadenylation signal, and a p-element for random genomic
		integration. Such a plasmid may have some 10 000 base pairs.<Br>

		After cloning, the picked clones must be verified, since the enzymes may
		not cut their target DNA completely, so that a recipient plasmid can be easily
		religated. This is especially true for recipient plasmids with a multiple cloning
		site as target and a short cutout fragment so that the incompletely digested
		recipient cannot be separated from the completely digested recipient plasmid
		on an agarose gel. Clones must also be verified for cloning strategies where the
		donor DNA can be inserted in both direction, either because the overhangs are
		compatible for that or because it is blunt end ligation.<Br>

		Here we provide a free online tool to find fitting enzymes for distinguishing
		between different DNA sequences. Simply give up to three sequences you want to
		distinguish, and we	will give you the restriction pattern for suitable enzymes
		with the most dissimilar pattern first.

		<div class="h2">Data security:</div>
			No sequence data will be uploaded, everything is processed in your local browser with JavaScript.
		</div>

		<div id="documentation" class="w3-container  tab" style="display: none;">

		<div class="h1">Introduction</div>

		In molecular cloning, restriction enzymes are used to cut out a fragment of DNA from a Donor Vector to
		ligate it into a previously opened Recipient Vector. The DNA fragments are separated on an agarose gel.
		However, restriction enzymes may digest incompletely, leaving not only backbone and insert, but also linearized DNA molecules in the reaction.
		A linearized plasmid of 10000 bp can be easily separated from a 1000 bp cutout fragment but is difficult to separate from a 9000 bp 
		backbone fragment. The problem is even worse if the cut out fragment is just a few basepairs from a multiple cloning site. Therefore, even after gel isolation linearized fragments may be left, and used in the ligation.
		A single fragment may be ligated with itself, than two are ligated together. But also cloning strategies where the overhangs are compatible for both orientations orientations including blunt end cloning require checking the identity of the picked clones.

		The clones can be identified by their restriction patterns in a text digest, easily. The challenge here is finding the right enzymes. An easy way to find such enzymes is provided by diffdigester.uni-jena.de.

		<figure>
			<img src="diffdigest.svg" alt="Flow Chart of a taypical procedure">
				<figcaption>Figure 1: A typical procedure in molecular cloning that ends a with a differential digest to identify a clone by the restriction pattern with a suitable enzyme. Diffdigester finds the such enzymes for testing your clones.</figcaption>
			</img>
		</figure>

		<br clear="left">

		You have to provide two or three sequences in plain text of fasta format. It does not matter whether you sequence is in upper or lower case, it will be converted to lower case for internal use. You can copy and paste the sequences into the web form or load them from a file.
		You may compare the original recipient, the insert and the final plasmid.

		An example of a plain text DNA sequence in 5-3 direction:
		AGCTAGTCGATGCTATATTTAGCGATGCTAGTCGACTAGCTAAGGCTAGTGATCGAGCTGATCTATCGGATCTAGCGATCGATCGACTAC<br>

		<div class="h1">What do you get?</div>

		You enter two or three DNA sequences, and indicate whether they are linear or circular. You get a gel with the restriction band patterns for each restriction enzyme that cuts your DNA sequences according to the selected settings. If you order "By dissimilarity", which is the default, the first enzymes will be those with most distinguishable patterns so that you can pick the best enzyme for your test digest to verify your clones.

		<div class="h1">How does it work?</div>

		The sequences you provide are processed with JavaScript. No sequence is uploaded to the server. All sequences remain in your browser.
		The JavaScript program uses regular expressions to search for the recognition sequences of restriction enzymes from the Rebase database. The resulting digest patterns differ by the length of the DNA fragments, which is a one dimensional value and thus we score the patterns with the Shortest Path Root Measure and sort them afterwards.

		<div class="h1">Disclaimer</div>

		For research use only. Not for diagnostic or therapeutic applications.

		<div class="h1">Literature</div>

		<div class="h2">About cloning, restriction enzymes, and their application:</div>
		<ul>
			<li>Matsumura, I. (2015) Why Johnny can’t clone: Common pitfalls and not so common solutions. BioTechniques, 59, 4-13.</li>
			<li>Zeng, Q., Eidsness, M.K., Summers, A.O. (1997) Near-Zero Background Cloning of PCR Products. BioTechniques, 23, 412-418.</li>
			<li>Spear, M.A. (2000) Efficient DNA subcloning through selective restriction endonuclease digestion. BioTechniques, 28, 660-668.</li>
			<li>Xu, Q., Zhang, D. and Downie, B. (2005) Rapid and Efficient Subcloning of DNA Without Dephosphorylation or Gel Electrophoresis. Applied Biochemistry and Biotechnology - Part B Molecular Biotechnology, 29, 111-117.</li>
			<li>Nathans, D. and Smith, H.O. (1975) Restriction endonucleases in the analysis and restructuring of dna molecules. Annu Rev Biochem, 44, 273-293.</li>
			<li>Roberts, R.J., Vincze, T., Posfai, J., Macelis, D. (2023) REBASE: a database for DNA restriction and modification: enzymes, genes and genomes Nucleic Acids Res, 51 (D1):D629-D630</li>
		</ul>
		<div class="h2">About measuring one-dimensional diversity:</div>
		<ul>
			<li>Enflo K.  Measuring one-dimensional diversity. Inquiry 2022; <a href="https://doi.org/10.1080/0020174X.2022.2125436">10.1080/0020174X.2022.2125436</a>.</li>
		</ul>
		<div class="h2">About design of plasmid vectors in general:</div>
		<ul>
			<li>Tolmachov O. Designing plasmid vectors. Methods Mol Biol. 2009;542:117-29.</li>
		</ul>

		<div class="h2">About p-vector cloning:</div>
		<ul>
			<li>Townson, S.M., Chang, B.S., Salcedo, E., Chadwell, L.V., Pierce, N.E. and Britt, S.G. (1998) Honeybee blue- and ultraviolet-sensitive opsins: cloning, heterologous expression in Drosophila, and physiological characterization. J Neurosci, 18, 2412-2422.</li>
			<li>Knox, B.E., Salcedo, E., Mathiesz, K., Schaefer, J. Chou, W.H., Chadwell, L.V., Smith, W.C., Britt, S.G., Barlow, R.B. (2003) Heterologous expression of Limulus rhodopsin. Journal of Biological Chemistry, 278, 40493-40502</li>
		</ul>
	</div>

	<div id="help" class="w3-container tab" style="display: none;">

		<h2>Quick Guide</h2>
		<ol class="arabic simple">
			<li><p>Load your DNA sequences from a file or paste them into the fields, and check the associated box if the sequence is circular. As an example, sequence 1 may be the sequence of the recipient plasmid, sequence 2 of the recipient plasmid, and sequence 3 of the final plasmid with the insert if the cloning strategy is designed so that all three could be picked as clone. The sequences can be raw or in fasta format and also be quite long since they are not uploaded to the server, but processed in the local browser. If more then one fasta sequence is in one text field, the first one is used and the others are ignored. The sequences should only contain the four canonical bases A, C, G, and T, case is ignored. Characters that are not letters in the English alphabet are ignored. Sequences for instance may come with a base pair index on the start of every new line. If a sequence contains another letter then A, C, G, or T, a warning is displayed below the “Submit and digest” button on submission. We recommend fixing this problem, since DiffDigester cannot know what to do with an ambiguous base such as N.</p></li>
		</ol>
		<a class="reference internal image-reference" href="docs/images/SequenceTextBoxes.png"><img alt="The text boxes to select the sequences." src="docs/images/SequenceTextBoxes.png"  style="width: 1200px" /></a>
		<ol class="arabic simple" start="2">
			<li><p>The settings allow you to limit the enzymes that are shown. The option "Minimum number of fragments" only shows enzymes that give at least the minimum number of fragments for each given sequence. This allows to exclude enzymes that do not cut any of the given sequences. The option "Maximum number of fragments" shows only enzymes that give at most the maximum number of fragments for each sequence. This allows to keep the number of fragments down as patterns with to many fragments are hard to recognize and distinguish. The options minimum and maximum fragment sizes restrict the results shown to enzymes that only give fragments whose sizes are within the given values. The option "Minimum fragment size on gel" allows to cut the gel at the given size in basepairs. Since the sizes are displayed logarithmically, there is otherwise a lot of empty space shown. Note that this option hides bands that are smaller, if the "Minimum fragment size" is smaller. In the image below are the default values show for the options.</p></li>
		</ol>
		<a class="reference internal image-reference" href="docs/images/Settings.png"><img alt="The settings" src="docs/images/Settings.png" /></a>
		<ol class="arabic simple" start="3">
			<li><p>The results can be shown in different orders, which are shown in the image below. They are "By Dissimilarity", "By Similarity", "Alphabetical" and "Reverse alphabetical". Since a diff digestion should differentiate between two DNA species the "By Dissimilarity" option is the default to show the most dissimilar digestion pattern first. Though reversing the order might also be useful, or just ordering the results alphabetical by enzyme name.</p></li>
		</ol>
		<a class="reference internal image-reference" href="docs/images/SortOrder.png"><img alt="The settings" src="docs/images/SortOrder.png" /></a>
		<ol class="arabic simple" start="4">
			<li><p>Choose the restriction enzymes you want to use for searching. For instance if you want to exclude some enzymes, select them in the “Enzymes to use” list box and move them with the arrow over to the “Enzyme to exclude” list box. You can use the shift and CTRL (CMD on a Mac) keys to select more than one enzyme at once. Alternatively, you can provide a text file containing the enzymes (format in the right panel). The file should contain one enzyme per row, and unknown enzymes will be ignored. DiffDigester will warn about the first unknown enzyme, it encounters. You can ignore this warning safely. In addition, you can save the list of enzymes for reuse or documentation.</p></li>
		</ol>
		<a class="reference internal image-reference" href="docs/images/EnzymeSelector.png"><img alt="Select the enzymes" src="docs/images/EnzymeSelector.png" style="width: 800px;" /></a>
		<ol class="arabic simple" start="5">
			<li><p>Click “Submit and Digest” to run the analysis. DiffDigester checks your inputs for the correct format and shows a warning if any characters in the English alphabet other than ‘ATGC’ are detected (arrow in the figure below). Characters that are not part of the English alphabet are ignored, such as basepair indices at the beginning of a line.</p></li>
		</ol>
		<a class="reference internal image-reference" href="docs/images/Warning.png"><img alt="Submit your DiffDigest" src="docs/images/Warning.png" style="width: 800px;" /></a>
		<ol class="arabic simple" start="6">
			<li><p>Your results will be displayed as mock gels in cartoon style for each enzyme that produces fragments according to the settings. Below is a gel for the enzyme TstI, the sequences used are from Example 1, which can be loaded from the Load example buttons. The first lane contains a DNA ladder with black bands, which are annotated with their sizes in basepairs. The second lane contains the bands produced from sequence 1 in red; the third lane bands from sequence 2 in green; and the fourth lane bands from sequence 3 in blue. Note that the minimum fragment size that was used for the gel in the image below was 50 bp, however this TstI also produces a fragment that is smaller then 50 bp and thus is not displayed. If a sequence is circular that means not cut then it is displayed as it were cut.</p></li>
		</ol>
		<a class="reference internal image-reference" href="docs/images/GelPic.png"><img alt="An example of an output gel in cartoon style" src="docs/images/GelPic.png" /></a>

		<h2>Examples</h2>
		<div class="section" id="example-1">
			<h3>Example 1: Replacing a fluorophore gene by another gene of interest and an oligo adapter.</h3>
			<p>This cloning example is based on replacing a fluorophore gene by another gene of interest and an adapter oligo to bridge the non-compatible restriction enzymes XhoI and HindIII. Recipient and donor plasmid are digested with two different combinations of restriction enzymes to generate fragments for the ligation procedure. PspOMI and NotI generate compatible overhangs, which is not true for XhoI and HindIII. For that reason, an adapter oligo with compatible overhangs for both enzymes was added during ligation to get the final plasmid. Out tool suggests the enzyme NcoI to create digestion patterns of all three plasmid (R, D, F), that are easily distinguishable. In comparison to the predicted digestion pattern, the agarose gel image is displayed.</p>
			<a class="reference internal image-reference" href="docs/images/diffdigest_example_01.svg"><img alt="example 1" src="docs/images/diffdigest_example_01.svg" style="width: 800px;" /></a>

			<div class="section" id="sequences1">
				<h4>Sequences:</h4>
				<details>
					<summary><a>Sequence 1 (recipient vector, 8115 bp)</a></summary>
					<div class="highlight-none notranslate">
						<div class="highlight">
							<pre><span></span><!-- Insert DNA --></pre>
						</div>
					</div>
				</details>
				<details>
					<summary><a>Sequence 2 (donor vector, 6566 bp)</a></summary>
					<div class="highlight-none notranslate">
						<div class="highlight">
							<pre><span></span></pre>
						</div>
					</div>
				</details>
				<details>
					<summary><a>Sequence 3 (final vector, 8674 bp)</a></summary>
					<div class="highlight-none notranslate">
						<div class="highlight">
							<pre><span></span></pre>
						</div>
					</div>
				</details>
				<details>
					<summary><a>Enzyme for differential digest: NcoI</a></summary>
				</details>

				<div class="line-block">
					<div class="line"><br /></div>
				</div>
			</div>
		</div>

		<div class="section" id="example-2">
			<h3>Example 2: Replacing a fluorophore gene by another fluorophore gene amplified by PCR.</h3>
			<p>The recipient plasmid, encoding EGFP, and the PCR product of the fluorophore tdTomato are digested by the restriction enzymes XhoI and XbaI. The ligation of both fragments results in the final tdTomato encoding plasmid. A digestion of isolated bacterial clones with AvaI offers the opportunity to identify correct clones by their digestion pattern.</p>
			<a class="reference internal image-reference" href="docs/images/diffdigest_example_02.svg"><img alt="example 2" src="docs/images/diffdigest_example_02.svg" style="width: 800px;" /></a>

			<div class="section" id="sequences2">
				<h4>Sequences:</h4>
				<details>
					<summary><a>Sequence 1 (recipient vector, 6160 bp)</a></summary>
					<div class="highlight-none notranslate">
						<div class="highlight">
							<pre><span></span></pre>
						</div>
					</div>
				</details>
				<details>
					<summary><a>Sequence 2 (final vector, 6882 bp)</a></summary>
					<div class="highlight-none notranslate">
						<div class="highlight">
							<pre><span></span></pre>
						</div>
					</div>
				</details>
				<details>
					<summary><a>Enzyme for differential digest: AvaI</a></summary>
				</details>

				<div class="line-block">
					<div class="line"><br /></div>
				</div>
			</div>
		</div>

	</div>

	<div id="about" class="w3-container tab" style="display: none;">
		<div class="h1">This work has been delveloped in the Mrowka Lab</div>
		<div>Source Code: <a href="https://github.com/MartinGuehmann/diffdigester">https://github.com/MartinGuehmann/diffdigester</a></div>

		<BR>
		Contact:<BR>
		Prof. Dr. Ralf Mrowka<BR>
		Experimental Nephrology<BR>

		<a href="https://www.uniklinikum-jena.de">Universitätsklinikum Jena</a>
		, KIM III<BR>
		Am Nonnenplan 4<BR>
		07743 JENA<BR>
		Germany<BR>

		Tel.: +49 3641 9396600 (Office)<BR>
		FAX:  +49 3641 9396602
		<BR>
		<BR>

		<div class="h2">
			<a href="https://www.uniklinikum-jena.de/Impressum.html">Imprint</a>
		</div>
	</div>

	<div id="digester" class="w3-container  tab" style="display: none;">

		<div class="h1">Diff digest your sequences</div><br>
		<input type="file" id="loadSequenceDialog" style="display:none" onchange="onChangeLoadSequences()"/>

		<div>DNA Sequence 1 (e.g. DNA sequence of recipient plasmid):
			<div>
				<textarea id="seq1" name="sequence 1" rows="4" cols="90"></textarea>
			</div>
			<div>
				<button class="button button2" onclick="loadSequence('seq1')" title="Load sequence from a file.">Load sequence...</button>
				<button class="button button5" onclick="setseq('seq1',sampleseq1)" title="Load the sequence of the recipient plasmid of example 1 from the example section.">Load example</button>
				<button id="clear1" class="button button3" onclick="clearseq('seq1')" title="Clear the current sequence and the result columns.">Clear</button>
				<label for="circular1" title="Check if the provided sequence is circular.">circular: </label><input id="circular1" checked="checked" type="checkbox" title="Check if the provided sequence is circular." />
			</div>
		</div><br>

		<div>DNA Sequence 2 (e.g. donor plasmid):
			<div>
				<textarea id="seq2" name="Sequence 2" rows="4" cols="90"></textarea>
			</div>
			<div>
				<button class="button button2" onclick="loadSequence('seq2')" title="Load sequence from a file.">Load sequence...</button>
				<button class="button button5" onclick="setseq('seq2',sampleseq2)" title="Load the sequence of the final plasmid of example 1 from the example section.">Load example</button>
				<button id="clear2" class="button button3" onclick="clearseq('seq2')" title="Clear the current sequence and the result columns.">Clear</button>
				<label for="circular2" title="Check if the provided sequence is circular.">circular: </label><input id="circular2" checked="checked" type="checkbox" title="Check if the provided sequence is circular." />
			</div>
		</div><br>

		<div>DNA Sequence 3 (e.g. final plasmid):
			<div>
				<textarea id="seq3" name="Sequence 3" rows="4" cols="90"></textarea>
			</div>
			<div>
				<button class="button button2" onclick="loadSequence('seq3')" title="Load sequence from a file.">Load sequence...</button>
				<button class="button button5" onclick="setseq('seq3',sampleseq3)" title="Load the sequence of the donor plasmid of example 1 from the example section.">Load example</button>
				<button id="clear3" class="button button3" onclick="clearseq('seq3')" title="Clear the current sequence and the result columns.">Clear</button>
				<label for="circular3" title="Check if the provided sequence is circular.">circular: </label><input id="circular3" checked="checked" type="checkbox" title="Check if the provided sequence is circular." />
			</div>
		</div><br>

		<div>
			<table style="text-align: left; border:none;" cellpadding="2" cellspacing="2">
				<tbody>
					<tr>
						<td style="vertical-align: top; border:none;">
							<label for="minFragments" title="Only show enzymes that cut each sequence at least into so many fragments.">Minimum number of fragments: </label>
						</td>
						<td style="horizontal-align: top; border:none;">
							<input id="minFragments" value="2" min="0" type="number" title="Only show enzymes that cut each sequence at least into so many fragments." />
						</td>
					<tr/>
					<tr>
						<td style="vertical-align: top; border:none;">
							<label for="maxFragments" title="Only show enzymes that cut each sequence at most into so many fragments.">Maximum number of fragments: </label>
						</td>
						<td style="horizontal-align: top; border:none;">
							<input id="maxFragments" value="10" min="0" type="number" title="Only show enzymes that cut each sequence at most into so many fragments." />
						</td>
					<tr/>
					<tr>
						<td style="vertical-align: top; border:none;">
							<label for="minFragmentSize" title="Only show enzymes that produce fragments bigger than that or equal.">Minimum fragment size (in bp): </label>
						</td>
						<td style="horizontal-align: top; border:none;">
							<input id="minFragmentSize" value="50" min="0" type="number" title="Only show enzymes that produce fragments bigger than that or equal." />
						</td>
					<tr/>
					<tr>
						<td style="vertical-align: top; border:none;">
							<label for="maxFragmentSize" title="Only show enzymes that produce fragments smaller than that or equal.">Maximum fragment size (in bp): </label>
						</td>
						<td style="horizontal-align: top; border:none;">
							<input id="maxFragmentSize" value="10000" min="0" type="number" title="Only show enzymes that produce fragments smaller than that or equal." />
						</td>
					<tr/>
					<tr>
						<td style="vertical-align: top; border:none;">
							<label for="gelBottom" title="Cut off the gel at fragment size smaller than this. This correspondents to fragments that have run through the gel and left it at the bottom. This option saves space on the page.">Minimum fragment size on gel (in bp): </label>
						</td>
						<td style="horizontal-align: top; border:none;">
							<input id="gelBottom" value="50" min="0" type="number" title="Cut off the gel at fragment size smaller than this. This correspondents to fragments that have run through the gel and left it at the bottom. This option saves space on the page." />
						</td>
					<tr/>
				</tbody>
			</table>
		</dif><br>

		<div>
			<div class="h2">Sort Order</div>
			<table style="text-align: left; border:none;" cellpadding="2" cellspacing="2">
				<tbody>
					<tr>
						<td style="vertical-align: top; border:none;">
							<input id="patternDissimilarity" value="0" name="SortOrder" type="radio" checked title="Sort the output by enzyme band pattern dissimilarity (most dissimular first)." />
						</td>
						<td style="horizontal-align: top; border:none;">
							<label for="patternDissimilarity" title="Sort the output by enzyme band pattern dissimilarity (most dissimular first).">By Dissimilarity</label>
						</td>
					<tr/>
					<tr>
						<td style="vertical-align: top; border:none;">
							<input id="patternSimilarity" value="1" name="SortOrder" type="radio" title="Sort the output by enzyme band pattern similarity (most simular first)." />
						</td>
						<td style="horizontal-align: top; border:none;">
							<label for="patternSimilarity" title="Sort the output by enzyme band pattern similarity (most simular first).">By Similarity</label>
						</td>
					<tr/>
					<tr>
						<td style="vertical-align: top; border:none;">
							<input id="alphabetical_A-Z" value="2" name="SortOrder" type="radio" title="Sort the output by enzyme name in alphabetical order (A - Z)." />
						</td>
						<td style="horizontal-align: top; border:none;">
							<label for="alphabetical_A-Z" title="Sort the output by enzyme name in alphabetical order (A - Z).">Alphabetical (A - Z)</label>
						</td>
					<tr/>
					<tr>
						<td style="vertical-align: top; border:none;">
							<input id="alphabetical_Z-A" value="3" name="SortOrder" type="radio" title="Sort the output by enzyme name in reverse alphabetical order (Z - A)." />
						</td>
						<td style="horizontal-align: top; border:none;">
							<label for="alphabetical_Z-A" title="Sort the output by enzyme name in reverse alphabetical order (Z - A).">Reverse alphabetical (Z - A)</label>
						</td>
					<tr/>
				</tbody>
			</table>
		</dif><br>

		<table style="text-align: left; border:none;" cellpadding="2" cellspacing="2">
			<tbody>
				<tr>
					<td style="vertical-align: top; border:none;">
						<label for="EnzymesToUse">Enzymes to use:</label>
					</td>
					<td style="horizontal-align: top; border:none;">
					</td>
					<td style="vertical-align: top; border:none;">
						<label for="EnzymesToExclude">Enzymes to exclude:</label>
					</td>
					<td style="horizontal-align: top; border:none;">
					</td>
				<tr/>
				<tr>
					<td style="vertical-align: top; border:none;">
						<div>
							<select id="EnzymesToUse" name="EnzymesToUse" style="width:180px; height:120px" size="6" multiple>
							</select>
						</div>
					</td>
					<td style="horizontal-align: top; border:none;">
						<div><button id="addEnzymes" class="button button2" onclick="addEnzymes()" title="Add one or more enzymes. Use 'shift' or 'control' (on a Mac 'cmd') to slect more than one enzyme.">&larr;</button></div>
						<div><button id="removeEnymes" class="button button3" onclick="removeEnzymes()" title="Remove one or more enzymes. Use 'shift' or 'control' (on a Mac 'cmd') to slect more than one enzyme.">&rarr;</button></div>
					</td>
					<td style="vertical-align: top; border:none;">
						<div>
							<select id="EnzymesToExclude" name="EnzymesToExclude" style="width:180px; height:120px" size="6" multiple>
							</select>
						</div>
					</td>
					<td style="horizontal-align: top; border:none;">
						<div>
							<button id="loadEnymes" class="button button2" onclick="loadEnzymes()" title="Load enzymes from a file.">Load Enzymes...</button>
							<input type="file" id="loadEnzymesDialog" style="display:none" onchange="onChangeLoadEnzymes()"/>
						</div>
						<div>
							<button id="saveEnymes" class="button button1" onclick="saveEnzymes()" title="Save enzymes to a file.">Save Enzymes...</button>
							<a id="saveEnzymesDialog" style="display:none"/>
						</div>
					</td>
					<td style="horizontal-align: top; border:none;">
						<div class='label' id='fileWarning' style="color:red"></div>
					</td>
				</tr>
				<tr>
					<td style="vertical-align: top; border:none;">
						<div><button id="useAllEnzymes" class="button button2" onclick="useAllEnzymes()" title="Use all enzymes in the database for digestion.">Use all enzymes</button></div>
					</td>
					<td style="horizontal-align: top; border:none;">
					</td>
					<td style="vertical-align: top; border:none;">
						<div><button id="removeAllEnymes" class="button button3" onclick="removeAllEnzymes()" title="Remove all enzymes from the list of enzymes to use. Then you can put back a few enzymes.">Remove all enzymes</button></div>
					</td>
					<td style="horizontal-align: top; border:none;">
					</td>
				<tr/>
			</tbody>
		</table><br>

		<div>
			<button class="button button4" onclick="myFunction()" title="When you have entered your sequences and the selected the enzymes you want to use, find the enzymes for your digest in the gels below.">Submit and Digest</button>
			<button id="clearAll" class="button button3" onclick="clearresults()" title="Clear the result columns.">Clear Results</button>
			<button id="clearSequences" class="button button3" onclick="clearSequences()" title="Clear all provided sequences and the result columns.">Clear All Sequences</button>
		</div><br>

		<div class='label' id='commenton'>Info: </div>
		<div id="ColorLegend" style="display: none;">Bands in <font color="red">red</font> are fragments from sequence 1; in <font color="green">green</font> from sequence 2; and in <font color="blue">blue</font> from sequence 3.</div>
		<br>
		<div class="h1" id="NoFragmentMessage" style="display: none;color:red;">No fragments found for the selected enzymes and settings. You may select other enzymes or change the settings for fragment numbers and sizes.</div>
		<div class="h1" id="NoSequence" style="display: none;color:red;">Please enter at least one sequence.</div>
		<div id="gelDrawings"><!-- For JavaScript to insert canvases --></dif>
	</div>
</body>
</html>
